xgi/generators/random.py

```diff
@@ -76,13 +82,74 @@ def random_hypergraph(N, ps, order=None, seed=None):
         else:
             d = i + 1  # order, ps[0] is prob of edges (d=1)
 
-        potential_edges = combinations(nodes, d + 1)
-        n_comb = comb(N, d + 1, exact=True)
-        mask = np.random.random(size=n_comb) <= p  # True if edge to keep
+        size = d + 1
+        if p > 0:
+            h_uniform = uniform_erdos_renyi_hypergraph(n, size, p, seed=seed)
+            H << h_uniform
+
+    return H
+
+
+def random_hypergraph(N, ps, seed=None):
```

Comment:
```suggestion
def random_hypergraph(n, ps, seed=None):
```

for consistency with the others